src/lib/server/main_loop.c

```diff
@@ -255,9 +255,8 @@ int main_loop_init(void)
 
 #ifdef HAVE_SYSTEMD_WATCHDOG
 	if (sd_watchdog_interval.tv_sec || sd_watchdog_interval.tv_usec) {
-		struct timeval	now;
-		fr_time_to_timeval(&now, fr_event_list_time(event_list));
-		sd_watchdog_event(event_list, &now, NULL);
+		fr_time_t now = fr_time();
+		sd_watchdog_event(event_list, now, NULL);
```

Comment:
I'm not sure you can change the `now` variable from `struct timeval` to `fr_time_t` like that.  What is `sd_watchdog_event()` expecting?